### PR TITLE
Update GitHub Action workflows to use aepsdk-commons

### PR DESCRIPTION
**Pull request by timkimadobe**
## Description

This PR migrates the local GitHub Action workflows to use the reusable workflows available in aepsdk-commons. At a high level:
* The reusable workflows are tagged per platform in aepsdk-commons (ex: `gha-android-1.0.0`), which the workflows use for controlling which version of the workflow is used
* The same style of manual workflow inputs has been generally preserved, and the goal is to abstract away the usage of the reusable workflows

`update-version.yml`
* The optional dependency list has expanded based on all of the existing dependencies in gradle.properties today

`maven-release.yml`
* Edge Identity dependency version has been added as a validation requirement for the release

> [!NOTE]
> Due to current GitHub Actions YAML limitations, YAML anchors are not available, meaning that the workflow tag and input value must be maintained simultaneously. However, they are planning on adding this in 2025.
> Source: https://github.com/actions/runner/issues/1182

Update versions (no dependencies):
* Job run: https://github.com/timkimadobe/aepsdk-edge-android/actions/runs/11490724851/job/31981920703
* Resulting PR: https://github.com/timkimadobe/aepsdk-edge-android/pull/3

Update versions (dependencies AND targeting different branch `dev` - demonstrating you can update versions in any branch):
* Job run: https://github.com/timkimadobe/aepsdk-edge-android/actions/runs/11490733780
* Resulting PR: https://github.com/timkimadobe/aepsdk-edge-android/pull/4

Publish release (failed because no secrets for publishing available in fork):
* Job run: https://github.com/timkimadobe/aepsdk-edge-android/actions/runs/11490639786/job/31981684268

Publish snapshot (failed because no secrets for publishing available in fork):
* Job run: https://github.com/timkimadobe/aepsdk-edge-android/actions/runs/11490687754/job/31981819389

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: .github/workflows/maven-release.yml [hunk 1]
```diff
@@ -15,62 +15,29 @@ on:
...
-        description: 'Create tag? ("no" to skip)'
+      create-github-release:
+        description: 'Create GitHub release with tag? ("false" to skip)'
+        type: boolean
         required: true
-        default: 'yes'
+        default: true
+
+      edge-identity-dependency:
+        description: 'The AEPEdgeIdentity dependency version in gradle.properties to be validated (ex: 3.0.1).'
```
Comment (addb) [inline .github/workflows/maven-release.yml:28]: nit: Here we use ex: 3.0.1 the [L18 tag](https://github.com/adobe/aepsdk-edge-android/pull/171/files#diff-1e1110e4b87ab2617e8052089ef9e5b403618114a3cf009ead9b3925d3bc89aaR18) has ex: 5.0.1 
 
 let's keep example versions also consistent and probably to latest major version i.e. 3.x

File: .github/workflows/update-version.yml [hunk 2]
```diff
@@ -1,48 +1,54 @@
...
+      contents: write
+      pull-requests: write
+    uses: adobe/aepsdk-commons/.github/workflows/versions.yml@version-script-update
+    with:
+      version: ${{ github.event.inputs.version }}
+      branch: ${{ github.event.inputs.branch }}
+      dependencies: AEPCore ${{ github.event.inputs.core-dependency }}, AEPEdgeIdentity ${{ github.event.inputs.edge-identity-dependency }}, AEPEdgeConsent ${{ github.event.inputs.edge-consent-dependency }}, AEPTestUtils ${{ github.event.inputs.testutils-dependency }}
+      paths: code/edge/src/main/java/com/adobe/marketing/mobile/EdgeConstants.java, code/gradle.properties
+      update: true
+      workflow_tag: gha-android-1.0.0
```
Comment (cacheung) [inline .github/workflows/update-version.yml:54]: Is this one supposed to be 'version-script-update'?

File: .github/workflows/update-version.yml [hunk 3]
```diff
@@ -1,48 +1,54 @@
...
+jobs:
+  update-versions:
+    permissions:
+      contents: write
+      pull-requests: write
+    uses: adobe/aepsdk-commons/.github/workflows/versions.yml@version-script-update
+    with:
+      version: ${{ github.event.inputs.version }}
+      branch: ${{ github.event.inputs.branch }}
+      dependencies: AEPCore ${{ github.event.inputs.core-dependency }}, AEPEdgeIdentity ${{ github.event.inputs.edge-identity-dependency }}, AEPEdgeConsent ${{ github.event.inputs.edge-consent-dependency }}, AEPTestUtils ${{ github.event.inputs.testutils-dependency }}
```
Comment (cacheung) [inline .github/workflows/update-version.yml:53]: I don't see iOS one has AEPEdgeConsent as dependency, while two platforms are different?

**Review by cacheung — APPROVED** [review]

File: .github/workflows/maven-release.yml [hunk 4]
```diff
@@ -15,62 +15,29 @@ on:
   workflow_dispatch:
     inputs:
       tag:
-        description: 'tag/version'
+        description: 'The existing tag (version) to be released (ex: 5.0.1).'
```
[the remark this answers is not in the document]
Comment (addb) [reply .github/workflows/maven-release.yml]: I was under the impression create GH release creates the tag and release. Isn't that the case?

File: .github/workflows/maven-release.yml (hunk 1 above)
Comment (addb) [reply .github/workflows/maven-release.yml]: works

File: .github/workflows/update-version.yml [hunk 5]
```diff
@@ -1,48 +1,56 @@
-#
```
Comment (kevinlind) [inline .github/workflows/update-version.yml:1]: Keep the copyright header for this file.

File: .github/workflows/update-version.yml [hunk 6]
```diff
@@ -1,48 +1,56 @@
...
+
+      branch:
+        description: 'The target branch where the version update will be applied and the pull request will be merged into.'
+        type: string
+        required: true
+        default: ''
 
       core-dependency:
-        description: '[Optional] Update Core dependency in pom.xml. Example: 3.0.0'
+        description: 'If a version is provided, update AEPCore dependency in gradle.properties (ex: 1.2.3).'
```
Comment (kevinlind) [inline .github/workflows/update-version.yml:19]: Specify that this is a dependency in the pom.xml file, so users know this is a published dependency.

File: .github/workflows/update-version.yml [hunk 7]
```diff
@@ -1,48 +1,56 @@
...
+        type: string
         required: false
+        default: ''
 
-      identity-dependency:
-        description: '[Optional] Update Edge Identity dependency in pom.xml. Example: 3.0.0'
+      edge-identity-dependency:
```
Comment (kevinlind) [inline .github/workflows/update-version.yml:24]: Specify that this is a dependency in the pom.xm file, so users know this is a publish dependency.

File: .github/workflows/update-version.yml [hunk 8]
```diff
@@ -1,48 +1,56 @@
...
+        description: 'If a version is provided, update AEPEdgeIdentity dependency in gradle.properties (ex: 1.2.3).'
+        type: string
         required: false
+        default: ''
 
-jobs:
-  update-version:
-    runs-on: ubuntu-latest
+      edge-consent-dependency:
+        description: 'If a version is provided, update AEPEdgeConsent dependency in gradle.properties (ex: 1.2.3).'
```
Comment (kevinlind) [inline .github/workflows/update-version.yml:31]: Specify this is a testing only dependency.

File: .github/workflows/update-version.yml [hunk 9]
```diff
@@ -1,48 +1,56 @@
...
+        default: ''
 
-    steps:    
-    - name: Checkout
-      uses: actions/checkout@v3
-    
-    - name: Update Edge
-      run: (./scripts/version.sh -u -v ${{ github.event.inputs.version }} -d "Core ${{ github.event.inputs.core-dependency }}, EdgeIdentity ${{ github.event.inputs.identity-dependency }}")
+      testutils-dependency:
+        description: 'If a version is provided, update AEPTestUtils dependency in gradle.properties (ex: 1.2.3).'
```
Comment (kevinlind) [inline .github/workflows/update-version.yml:37]: Specify this is a testing only dependency.

**Review by kevinlind — APPROVED** [review]

**Pull request merged**